sdk/test/Spec.hs

```diff
@@ -3,11 +3,14 @@ import qualified OpenTelemetry.ContextSpec as ContextSpec
 import qualified OpenTelemetry.ResourceSpec as ResourceSpec
 import OpenTelemetry.Trace (initializeGlobalTracerProvider)
 import qualified OpenTelemetry.TraceSpec as TraceSpec
+import System.Environment (setEnv)
 import Test.Hspec
 
 
 main :: IO ()
 main = do
+  -- For the attribute length limit test
+  setEnv "OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT" "50"
```

Comment:
Why doesn't the test itself set this? Do we need this action-at-a-distance?

Reply:
It isn't currently easy to set this locally. It gets read when we set up the global tracer provider, which we do on the next line, so it's not clear that we can do it at any other time.

Reply:
Ok let's add a comment both here, and where we need it

Reply:
Nvm I see you've done just that!